libraries/apollo-runtime/src/appleMain/kotlin/com/apollographql/apollo3/network/http/DefaultHttpEngine.apple.kt

```diff
@@ -320,3 +314,19 @@ private class Pipe {
     }
   }
 }
+
+interface DataTaskFactory {
+  fun dataTask(request: NSURLRequest): NSURLSessionDataTask
+}
+
+private class DefaultDataTaskFactory(delegate: NSURLSessionDataDelegateProtocol) : DataTaskFactory {
+  private val nsUrlSession = NSURLSession.sessionWithConfiguration(
+      configuration = NSURLSessionConfiguration.defaultSessionConfiguration(),
+      delegate = delegate,
+      delegateQueue = null
+  )
+
+  override fun dataTask(request: NSURLRequest): NSURLSessionDataTask {
+    return nsUrlSession.dataTaskWithRequest(request)
+  }
+}
```

Comment:
Thinking about this more, do we even need `DataTaskFactory`? Can't we "just" have 

```
actual class DefaultHttpEngine : HttpEngine {
  constructor(
      timeoutMillis: Long = 60_000,
      configuration: NSURLSessionConfiguration?
  ) {
```

What is the use case for changing anythign else than the configuration?

Reply:
I think we want to allow people to have a global `NSURLSession` for their app, like we do on the JVM with constructors that accept a `OkHttpClient` or `Call.Factory`.

But honestly it's hard to tell if it's actually useful without being an iOS dev 😅

Reply:
> I think we want to allow people to have a global NSURLSession for their app, like we do on the JVM with constructors that accept a OkHttpClient or Call.Factory.

Given we force the delegate, I don't think the `NSURLSession` can be reused by any other parts of the app so we might as well own it (also we're talking about closing it ourselves so if it is shared we def. don't want to close it and we'd have to refcount and what not)

Also looks like the iOS client allocates its own `URLsession`: https://github.com/apollographql/apollo-ios/blob/b434fb904b4695bcc84e83ca1cac132321bf5435/Sources/Apollo/URLSessionClient.swift#L64

Reply:
Good points, yeah that sounds about right. Simply passing a `NSURLSessionConfiguration` is definitely a welcome simplification 👍